src/workers/cdc/primary.ts

```diff
@@ -835,16 +837,32 @@ const onUserStateChange = async (
 const onSourceMemberChange = async (
   con: DataSource,
   logger: FastifyBaseLogger,
-  data: ChangeMessage<SourceMember>,
+  data: ChangeMessage<ContentPreferenceSource>,
 ) => {
+  const sourceId =
+    data.payload.after!.sourceId || data.payload.before!.sourceId;
+
+  const source = await con
```

Comment:
Maybe we should consider storing a flag that says on content preference that this is a squad? It will make fetching the source redundant and the content preference object will be complete

Reply:
I think it's a downgrade from what we had and might make it unclear for other devs that it needs to be queried

Reply:
Thinking, should we have gone with a route to have different type for squad `content_preference` then.

Reply:
Since it's water under the bridge now, let's proceed as is then. We should just be clear about it and pay attention